java/memory/memory-core/pom.xml

```diff
@@ -37,4 +37,58 @@
     </dependency>
   </dependencies>
 
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-surefire-plugin</artifactId>
+
+        <configuration>
+          <excludes>
+            <!-- Test is only useful when NOT running with add-opens -->
+            <exclude>**/TestOpens.java</exclude>
+          </excludes>
+        </configuration>
+      </plugin>
+    </plugins>
+  </build>
+
+  <profiles>
+    <profile>
+      <id>opens-tests</id>
+      <!-- Run tests WITHOUT add-opens to make sure we fail-fast -->
+      <activation>
+        <jdk>[16,]</jdk>
```

Comment:
Trying to get the TestOpens exception with JRE18 but if finished without that error, Is there some additional configuration needed? Probably `mvn -Dtest="TestOpens" clean test` not thrown that exception because for test this inherit [parent](https://github.com/apache/arrow/blob/59b57287e3e17e84ac41fdf7662fb2be2d4ef9f5/java/pom.xml#L842) configuration.

````
[ERROR] Failures: 
[ERROR]   TestOpens.testMemoryUtilFailsLoudly:32 Expected java.lang.Throwable to be thrown, but nothing was thrown.
````

Reply:
It works with just `mvn test`. I presume specifying the test name manually is causing it to bypass this configuration.

Reply:
Thank you, that was the case